rabbitmq-native/src/main/groovy/com/budjb/rabbitmq/connection/ConnectionContextImpl.groovy

```diff
@@ -131,10 +143,13 @@ class ConnectionContextImpl implements ConnectionContext {
             return
         }
 
+        applicationEventPublisher.publishEvent(new ConnectionContextStoppingEvent(this))
+
         connection.close()
         connection = null
 
```

Comment:
Something I missed in my PR: Does this method need to apply any cleanup with regard to the connection metrics? Maybe not, but I wasn't thinking that far ahead when I was writing that.